<launch>
  <arg name="base_name"         default="kinect2"/>
  <arg name="queue_size"        default="5"/>
  <arg name="machine"           default="localhost" />
  <arg name="nodelet_manager"   default="$(arg base_name)" />
  <arg name="start_manager"     default="true" />
  <arg name="use_machine"       default="true" />

  <machine name="localhost" address="localhost" if="$(arg use_machine)"/>
  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)" args="manager"
        if="$(arg start_manager)" machine="$(arg machine)" output="screen"/>

  <node pkg="nodelet" type="nodelet" name="$(arg base_name)_points_xyzrgb_sd"
        args="load depth_image_proc/point_cloud_xyzrgb $(arg nodelet_manager)" respawn="true">
    <remap from="rgb/camera_info"             to="$(arg base_name)/sd/camera_info" />
    <remap from="rgb/image_rect_color"        to="$(arg base_name)/sd/image_color_rect" />
    <remap from="depth_registered/image_rect" to="$(arg base_name)/sd/image_depth_rect" />
    <remap from="depth_registered/points"     to="$(arg base_name)/sd/points" />
    <param name="queue_size" type="int" value="$(arg queue_size)" />
  </node>
</launch>
